Week-3/Session-9/Number guessing/script.js

```diff
@@ -0,0 +1,32 @@
+const playGuessingGame = (numToGuess, totalGuesses = 10) => {
+    let promptText = 'Enter a number between 1 and 100.';
+    let userInput;
+    let numGuesses = 0;
+
+    while (numGuesses < totalGuesses) {
+        userInput = prompt(promptText);
+
+        if (userInput === null || userInput === '') {
+            return 0;
+        }
+
+        if (isNaN(userInput)) {
+            promptText = 'Please enter a number.';
+            continue;
+        }
+
+        var guess = parseInt(userInput, 10);
```

Comment:
Defer from using var here. Var is functional scoped, and hence won't be scoped inside of the while statement. One might run into overriding the value of guess outside of the while statement, hence making the output unpredictable.

Reply:
Noted @kamalmohit